schema.sql

```diff
@@ -13,258 +18,301 @@ drop table if exists areas;
-	primary key (prime_mover, fuel_type)
+create table  prime_movers (
+        id integer primary key
+        ,prime_mover text not null
+        ,fuel text null
+        ,description text null
+        ,unique (prime_mover, fuel)
 );
 
 -- create table for balancing topologies
 create table balancing_topologies (
-	name text not null primary key,
-	area text null references areas(name),
-	participation_factor float default 1.0 not null check (participation_factor >= 0 and participation_factor <= 1),
-	description text null
+        id integer primary key
+	,name text not null unique
+	,area text null references areas(name)
+	,participation_factor real default 1.0 not null check (participation_factor >= 0 and participation_factor <= 1)
+	,description text null
 );
 
 -- change to planning regions
 create table  areas (
-	name text not null primary key,
-	description text null
+        id integer primary key
+	,name text not null unique
+	,description text null
 );
 
--- electrical information of the lines
-create table  transmission_lines (
-	balancing_topology_from text references balancing_topologies(name),
-	balancing_topology_to text references balancing_topologies(name),
-	area_from text references areas(name),
-	area_to text references areas(name),
-	continuous_rating float not null check(continuous_rating >= 0),
-	ste_rating float not null check (ste_rating >=0),
-	lte_rating float not null check (lte_rating >=0),
-	line_length float not null check (line_length >= 0)
+-- physical connection between entities
+create table arcs(
+        id integer primary key
+        ,from_to int
+        ,to_from int
+        ,foreign key (from_to) references entities(id)
+        ,foreign key (to_from) references entities(id)
 );
 
+-- transmission lines
+create table transmission_lines (
+        id integer primary key
+        ,arc_id int
+	,continuous_rating real not null check(continuous_rating >= 0)
+	,ste_rating real not null check (ste_rating >=0)
+	,lte_rating real not null check (lte_rating >=0)
+	,line_length real not null check (line_length >= 0)
+        ,foreign key (arc_id) references arcs(id)
+) strict;
+
 
--- ) strict;
 -- flow between two regions
-create table  transmission_interchange (
-	area_from text not null references areas(name),
-	area_to text not null references areas(name),
-	max_flow_from float not null,
-	max_flow_to float not null
-);
+create table transmission_interchange (
+        arc_id int references arcs(id)
+        ,max_flow_from real not null
+	,max_flow_to real not null
+) strict;
 
--- create load input table, at some point need to add the growth rate
-create table  demand_requirements (
-	entity_attribute_id integer primary key,
-	peak_load float not null,
-	area text references areas(name),
-	balancing_topology text references balancing_topologies(name)
+-- create supplemental atttributes
+create table supplemental_attributes (
+	id integer primary key
+        ,type text not null
+	,value jsonb not null
```

Comment:
So supplemental_attributes are more like the `attributes` table we have before, but there is a many-to-many relationship between them instead.

When we are converting from Sienna PSY to the database or back, which attributes will have their own identity that we try and share and which attributes will be copied into different fields (like specific two-terminal loss field vs time-series). Is that going to be based on a constant list?

Reply:
I think the easiest would be that applications choose what fields they use to create the structs.